### PR TITLE
MISUV-8344 Filter PoA Charges if Coded Out

### PR DESCRIPTION
**Pull request by danielmaltz1**

### CODE REVIEW

File: app/controllers/TaxYearSummaryController.scala [hunk 1]
```diff
@@ -149,6 +149,13 @@ class TaxYearSummaryController @Inject()(taxYearSummaryView: TaxYearSummary,
     }
   }
 
+  private val notCodedOutPoa: TaxYearSummaryChargeItem => Boolean = chargeItem => {
```
Comment (avelx) [inline app/controllers/TaxYearSummaryController.scala:152]: if we can move this under ChargeItem and add some tests around this logic?

File: app/services/WhatYouOweService.scala [hunk 2]
```diff
@@ -43,6 +43,13 @@ class WhatYouOweService @Inject()(val financialDetailsService: FinancialDetailsS
     }
   }
 
+  val notCodedOutPoa: ChargeItem => Boolean = chargeItem => {
```
Comment (avelx) [inline app/services/WhatYouOweService.scala:46]: same as above, looks like a duplicate code which we can avoid by moving it in ChargeItem?

File: app/controllers/TaxYearSummaryController.scala (hunk 1 above)
Comment (danielmaltz1) [reply app/controllers/TaxYearSummaryController.scala:152]: TaxYearSummaryChargeItem is a different type to ChargeItem, but I can move both of these bits of code into the relevant type

File: app/models/financialDetails/ChargeItem.scala [hunk 3]
```diff
@@ -80,6 +80,13 @@ case class ChargeItem (
     subTransactionType.exists(subType => codingOutSubTypes.contains(subType))
   }
 
+  def notCodedOutPoa: Boolean = {
```
Comment (avelx) [inline app/models/financialDetails/ChargeItem.scala:83]: Please move this method from [ChargeItem] and [TaxYearSummaryChargeItem] to base trait => TransactionItem
here is example:
https://github.com/hmrc/income-tax-view-change-frontend/pull/2567/files

**Review by avelx — APPROVED** [review]

**Pull request merged**